app/src/main/java/pm/gnosis/heimdall/ui/qrscan/Videographer.kt

```diff
@@ -0,0 +1,182 @@
+package pm.gnosis.heimdall.ui.qrscan
+
+import android.app.Activity
+import android.graphics.ImageFormat
+import android.graphics.SurfaceTexture
+import android.hardware.Camera
+import android.os.Build
+import android.os.Handler
+import android.os.HandlerThread
+import android.view.Surface
+import android.view.TextureView
+import timber.log.Timber
+
+/*
+ * Check https://github.com/walleth/walleth/tree/master/app/src/main/java/org/walleth/activities/qrscan
+ */
+@Suppress("DEPRECATION")
+class Videographer(val activity: Activity) {
```

Comment:
We could use the context here and acquire the window manager as a system service:
https://developer.android.com/reference/android/view/WindowManager.html

This would allow us to inject this class